src/Extensions/Diagnostics.HealthChecks/ServiceCollectionExtensions.cs

```diff
@@ -30,20 +31,47 @@ public static IHealthChecksBuilder AddServiceFabricHealthChecks(this IServiceCol
 				});
 
 			serviceCollection.TryAddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
+		}
 
-			return serviceCollection
-				.AddHealthCheckPublisher<ServiceFabricHealthCheckPublisher>()
-				.AddHealthChecks();
+		/// <summary>
+		/// Register publisher for processing health check results directly to replicas
+		/// </summary>
+		public static IHealthChecksBuilder AddServiceFabricHealthChecks(this IServiceCollection serviceCollection)
+		{
+			return serviceCollection.AddServiceFabricHealthChecks<ServiceFabricHealthCheckPublisher>();
+		}
+
+		/// <summary>
+		/// Register publisher for processing health check results directly to nodes using REST api
+		/// </summary>
+		public static IHealthChecksBuilder AddRestHealthChecks(this IServiceCollection serviceCollection)
+		{
+			serviceCollection.TryAddSingleton(
+				provider =>
+				{
+					return provider.GetRequiredService<IOptions<RestHealthCheckPublisherOptions>>().Value;
+				});
```

Comment:
`IOptions<>` registered as open generic, there is no need to register each option separately.
```suggestion
```